lib/format_routing_constraint.rb

```diff
@@ -1,32 +1,27 @@
 class FormatRoutingConstraint
-  def initialize(format, content_format_inspector: ContentFormatInspector)
+  def initialize(format)
     @format = format
-    @content_format_inspector = content_format_inspector
   end
 
   def matches?(request)
-    @request = request
-    content_format == @format
+    content_item = set_content_item(request)
+    @format == content_item&.[]("schema_name")
   end
 
-private
+  def set_content_item(request)
```

Comment:
Nit: This method name is the same as [ApplicationController#set_content_item](https://github.com/alphagov/frontend/blob/adcddb8f8fff45e21c25b535baf7ba9c4c712e2b/app/controllers/application_controller.rb#L72), and is somewhat confusing as they do different things. (but probably worth renaming the other method instead... or this method could be something like `fetch_content_item` to make it more clear its requesting external data, rather than just setting an attribute)